File: Largest Number.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var largestNumber = function(nums) {
    
    for(let i=0; i<nums.length; i++){
        nums[i] = nums[i].toString();
    }
    
    
    let noSwap =true
    for(let i=nums.length; i>0; i--){
        noSwap = true;
        for(let j=0; j<i-1; j++){
            if(nums[j]+nums[j+1] < nums[j+1] + nums[j]){
                
                let temp =nums[j];
                nums[j] = nums[j+1]
                nums[j+1] = temp;
                noSwap=false;
            }
        }
        if(noSwap) break;
        
    }
    
    
    return nums[0] == 0? '0' : nums.join("");
    
    
};

File: Largest Number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        nums = list(map(str, nums))

        def compare(num1, num2):
            choice1 = num1 + num2
            choice2 = num2 + num1
            return int(choice2) - int(choice1)
        
        nums.sort(key = cmp_to_key(compare))
        
        if nums[0] == "0":
            return "0"
        return "".join(nums)
